### ISSUE TITLE
Automatic OkHttpEventListener breaks Pusher push notifications SDK initialization

### ISSUE THREAD

**Issue by sebj** [getsentry/sentry-android-gradle-plugin#513]
### Gradle Version

8.1.1

### AGP Version

7.4.2

### Code Minifier/Optimizer

None

### Version

3.11.0

### Sentry SDK Version

6.23.0

### Steps to Reproduce

1. Create a new project integrating the [Pusher push notifications SDK](https://github.com/pusher/push-notifications-android) as well as Sentry
2. Build and run the project on an Android device (debug build)
3. Observe the project crashes at launch

Note that the latest Sentry Java version (6.23.0) with the previous Gradle plugin (3.10.0) causes no issues. It's only updating the Gradle plugin that seems to cause an incompatibility. If you think the problem actually lies with Pusher, I'll be happy to open an issue with them! 😅 

### Expected Result

App does not crash

### Actual Result

App crashes, Logcat output:

```
FATAL EXCEPTION: main
Process: com.daybridge.android.staging, PID: 19296
java.lang.IllegalAccessError: Field 'okhttp3.OkHttpClient$Builder.eventListenerFactory' is inaccessible to class 'okhttp3.OkHttpClient' (declaration of 'okhttp3.OkHttpClient' appears in /data/app/~~YHzQ9oKGAXvDjzN6l2rICg==/com.daybridge.android.staging--VYirkni3VpiBa1pBR1Q7g==/base.apk!classes19.dex)
	at okhttp3.OkHttpClient.<init>(OkHttpClient.kt:121)
	at okhttp3.OkHttpClient$Builder.build(OkHttpClient.kt:1069)
	at com.pusher.pushnotifications.api.PushNotificationsAPI.<init>(PushNotificationsAPI.kt:94)
	at com.pusher.pushnotifications.PushNotificationsInstance$serverSyncHandler$1.invoke(PushNotificationsInstance.kt:108)
	at com.pusher.pushnotifications.PushNotificationsInstance$serverSyncHandler$1.invoke(PushNotificationsInstance.kt:93)
	at com.pusher.pushnotifications.PushNotificationsInstance.<init>(PushNotificationsInstance.kt:105)
	at com.pusher.pushnotifications.internal.PushNotificationsInitProvider.onCreate(ApplicationLifecycle.kt:67)
	at android.content.ContentProvider.attachInfo(ContentProvider.java:2451)
	at android.content.ContentProvider.attachInfo(ContentProvider.java:2421)
	at android.app.ActivityThread.installProvider(ActivityThread.java:7474)
	at android.app.ActivityThread.installContentProviders(ActivityThread.java:6985)
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6756)
	at android.app.ActivityThread.-$$Nest$mhandleBindApplication(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2129)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7884)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
```

**Comment by barry-irvine** [comment]
Got the same issue when running all my instrumentation tests after upgrading. (Using a HiltTestRunner if that makes any difference).

**Comment by sedubois** [comment]
There is a similar crash on init with the Intercom SDK after upgrading from 3.10.0 to 3.11.0.

**Comment by romtsn** [comment]
@stefanosiano do you wanna look into this or should I? I think this is a p0.5 as it's blocking people from using our latest SAGP version

**Comment by zmunm** [comment]
May I know deployment plan?

**Comment by romtsn** [comment]
> May I know deployment plan?

we'll ship it today

**Comment by kahest** [comment]
This was just released in [3.11.1](https://github.com/getsentry/sentry-android-gradle-plugin/releases/tag/3.11.1)

**Comment by sebj** [comment]
Thank you!
